From 2354d1d06380790f6b3871f549a20f80598cb1f8 Mon Sep 17 00:00:00 2001
From: Arturo Rinaldi <arty.net2@gmail.com>
Date: Wed, 16 Nov 2016 11:12:29 +0100
Subject: [PATCH] patches-4.4-925-MIPS-ath79-Linino-gpio-support

---
 arch/mips/ath79/gpio.c                         | 5 +++++
 arch/mips/include/asm/mach-ath79/ar71xx_regs.h | 3 +++
 2 files changed, 8 insertions(+)

# diff --git a/arch/mips/ath79/gpio.c b/arch/mips/ath79/gpio.c
# index 8f6fea5..4dfccd5 100644
# --- a/arch/mips/ath79/gpio.c
# +++ b/arch/mips/ath79/gpio.c
# @@ -62,10 +62,15 @@ static void __iomem *ath79_gpio_get_function2_reg(void)
#  void ath79_gpio_function2_setup(u32 set, u32 clear)
#  {
#  	void __iomem *reg = ath79_gpio_get_function2_reg();
# +	unsigned long flags;
# +
# +	spin_lock_irqsave(&ath79_gpio_lock, flags);
#
#  	__raw_writel((__raw_readl(reg) & ~clear) | set, reg);
#  	/* flush write */
#  	__raw_readl(reg);
# +
# +	spin_unlock_irqrestore(&ath79_gpio_lock, flags);
#  }
#
#  void ath79_gpio_function_setup(u32 set, u32 clear)
diff --git a/arch/mips/include/asm/mach-ath79/ar71xx_regs.h b/arch/mips/include/asm/mach-ath79/ar71xx_regs.h
index 050824f..352f3e2 100644
--- a/arch/mips/include/asm/mach-ath79/ar71xx_regs.h
+++ b/arch/mips/include/asm/mach-ath79/ar71xx_regs.h
@@ -973,6 +973,9 @@
 #define AR724X_GPIO_FUNC_JTAG_DISABLE		BIT(0)

 #define AR933X_GPIO_FUNC2_JUMPSTART_DISABLE 	BIT(9)
+#define AR933X_GPIO_FUNC2_WPS_DISABLE		BIT(8)
+#define AR933X_GPIO_FUNC2_I2SD_ON_12		BIT(5)
+#define AR933X_GPIO_FUNC2_DIS_MIC  	      	BIT(0)

 #define AR913X_GPIO_FUNC_WMAC_LED_EN		BIT(22)
 #define AR913X_GPIO_FUNC_EXP_PORT_CS_EN		BIT(21)
